src/test/java/core/basesyntax/ValidationTest.java

```diff
@@ -0,0 +1,264 @@
+        validUserBob.setLogin(VALID_LOGIN_FIRST);
+        validUserBob.setPassword(VALID_PASSWORD_FIRST);
+        validUserBob.setAge(VALID_AGE_FIRST);
+
+        validUserSecondBob = new User();
+        validUserSecondBob.setLogin(validUserBob.getLogin());
+        validUserSecondBob.setPassword(validUserBob.getPassword());
+        validUserSecondBob.setAge(validUserBob.getAge());
+
+        validUserAlice = new User();
+        validUserAlice.setLogin(VALID_LOGIN_SECOND);
+        validUserAlice.setPassword(VALID_PASSWORD_SECOND);
+        validUserAlice.setAge(VALID_AGE_SECOND);
+
+        validUserJohn = new User();
+        validUserJohn.setLogin(VALID_LOGIN_THIRD);
+        validUserJohn.setPassword(VALID_PASSWORD_THIRD);
+        validUserJohn.setAge(VALID_AGE_THIRD);
+
+        validUserData = new ArrayList<>();
+        validUserData.add(validUserBob);
+        validUserData.add(validUserAlice);
+        validUserData.add(validUserJohn);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_checkPrivateFields_Ok() {
+        List<Field> privateFields = new ArrayList<>();
+        List<Field> allFields = Arrays.asList(RegistrationServiceImpl.class.getDeclaredFields());
+        for (Field field : allFields) {
+            if (Modifier.isPrivate(field.getModifiers())) {
+                privateFields.add(field);
+            }
+        }
+        Assertions.assertEquals(allFields.size(), privateFields.size(),
+                "You should have private fields only!" + '\n');
+        Assertions.assertTrue(privateFields.size() >= allFields.size(),
+                "You should have at least " + allFields.size() + " fields" + '\n');
+    }
+
+    @Test
+    void register_ValidUsers_Ok() throws RegistrationException {
+        for (User user : validUserData) {
+            registrationService.register(user);
+        }
+
+        Assertions.assertEquals(validUserData.size(), Storage.people.size(),
+                "Test failed! The size of the storage isn't correct. Expected "
+                        + validUserData.size() + " but was " + Storage.people.size() + '\n');
+        Assertions.assertTrue(Storage.people.contains(validUserBob),
+                "Test failed! The storage must contain a user with login: "
+                        + validUserBob.getLogin() + '\n');
+        Assertions.assertTrue(Storage.people.contains(validUserAlice),
+                "Test failed! The storage must contain a user with login: "
+                        + validUserAlice.getLogin() + '\n');
+        Assertions.assertTrue(Storage.people.contains(validUserJohn),
+                "Test failed! The storage must contain a user with login: "
+                        + validUserJohn.getLogin() + '\n');
+    }
+
+    @Test
+    void register_AndGetValidUsers_Ok() throws RegistrationException {
+        for (User user : validUserData) {
+            registrationService.register(user);
+        }
```

Comment:
You have already successfully registered these users in the previous method. There is no need to have two test methods for the same user list.

Reply:
Can combine this test with the previous one? There I checked the existence of an object through contains, and not using a getter